bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -266,8 +276,24 @@ public void run() {
     return resultScanner;
   }
 
+  private CancellationToken createCancellationToken(final PooledChannel reservedChannel,
+      final ClientCall<ReadRowsRequest, ReadRowsResponse> readRowsCall) {
+    // If the scanner is close()d before we're done streaming, we want to cancel the RPC:
+    CancellationToken cancellationToken = new CancellationToken();
+    cancellationToken.addListener(new Runnable() {
+      @Override
+      public void run() {
+        readRowsCall.cancel();
+        if (reservedChannel != null) {
```

Comment:
The more variables we have to keep the less readable code.
How about this 

createCancellationToken( isGet ? null : channel, readRowsCall)
or
createCancellationToken( (channel instanceof  PooledChannel)? null : channel, readRowsCall)

some upfront cost.